llvm/lib/Target/AMDGPU/Utils/AMDGPUPALMetadata.cpp

```diff
@@ -232,8 +254,16 @@ void AMDGPUPALMetadata::setEntryPoint(unsigned CC, StringRef Name) {
   if (isLegacy())
     return;
   // Msgpack format.
+  // Entry point is updated to .entry_point_symbol and is set to the function
+  // name
   getHwStage(CC)[".entry_point_symbol"] =
       MsgPackDoc.getNode(Name, /*Copy=*/true);
+
+  // Set .entry_point which is defined
+  // to be _amdgpu_<stage> and _amdgpu_cs for non-shader functions
+  std::string EPName("_amdgpu_");
+  EPName += std::string(getStageName(CC) + 1);
```

Comment:
SmallString? also use the ostream to build the string

Reply:
Done